cmd/zfs/zfs_main.c

```diff
@@ -6120,8 +6120,12 @@ share_mount(int op, int argc, char **argv)
 		start_progress_timer();
 		get_all_datasets(&dslist, &count, verbose);
 
-		if (count == 0)
+		if (count == 0) {
+			if (options != NULL) {
+				free(options);
+			}
```

Comment:
[style] For single line conditionals the { } should not be added.  The same is true for the following hunks.